File: Check_if_given_liked_list_is_palindrome/code.cpp
#include <bits/stdc++.h>
bool isPalindrome(Node *head)
{
    // write your code here 
    if(head == NULL || head->next == NULL){
        return 1;
    }            
    else{
        vector<int> res;
        Node* ptr = head;
        while(ptr != NULL){
            res.push_back(ptr->data);
            ptr = ptr->next;
        }
        for(int i=0; i<res.size()/2; i++){
            if(res[i] != res[res.size()-i-1]){
                return 0;
            }
        }
        return 1;
    }        
}

File: Check_if_given_liked_list_is_palindrome/code.java
class Solution
{
    //Function to check whether the list is palindrome.
    boolean isPalindrome(Node head) 
    {
        //Your code here
        Node temp=head;int i=0;
        while(temp!=null)
        {
            i=i*10 + temp.data;
            temp=temp.next;
        }
        int z=i;
        int reverse = 0;  
        while(i!= 0)   
        {  
              int remainder = i % 10;  
              reverse = reverse * 10 + remainder;  
              i= i/10;  
         }   
         if(z==reverse)
         return true;
         else
         return false;

    }    
}
